www/auth/account.php

```diff
@@ -8,53 +8,43 @@
 
 require_once '../../lib/Util.php';
 
-$username = Util::getRequestParameter('username');
-$name = Util::getRequestParameter('name');
-$email = Util::getRequestParameter('email');
-$submitButton = Util::getRequestParameter('submitButton');
-
-// Determine whether we have a first-time login
-list($identity, $user) = Session::retrieveFirstLogin();
-
-if ($user) {
-  $firstLogin = true;
-  Util::requireNotLoggedIn();
-} else {
-  $firstLogin = false;
-  Util::requireLoggedIn();
-  $user = User::get_by_id(Session::getUser()->id); // cannot cache resource objects, need to refresh them
-}
-
+$name = Request::get('name');
+$email = Request::get('email');
+$newPassword = Request::get('newpassword');
+$newPassword2 = Request::get('newpassword2');
+$password= Request::get('password');
+$submitButton = Request::get('name');
+$errors=[];
+
+Util::requireLoggedIn();
+$user = User::get_by_id(Session::getUser()->id);
 // Save action
-if ($submitButton) {
-  if ($firstLogin) {
-    $user->username = $username;
-  }
+if ($submitButton && password_verify($password, $user->password)) {
   $user->name = $name;
   $user->email = $email;
-  if ($firstLogin && !$identity) {
-    FlashMessage::add(_('Could not find your OpenID; did you really log in?'));
```

Comment:
Agreed. There is more OpenID cleanup to do, since we ended up not using OpenID at all. For example, `templates/auth/beginAuth.tpl` can be removed entirely.